### ISSUE TITLE
[Buttons]: Squircle looks bad on our plain/plain-faint buttons with a background hover

### ISSUE THREAD

**Issue by fallaciousreasoning** [brave/leo#669]
@AlanBreck in our recent updates to the button shapes, I don't think we should've tweaked the background shape of the plain/plain-faint buttons as it makes a bunch of our existing buttons look weird:

Hovering over the close button for Brave News:
![image](https://github.com/brave/leo/assets/7678024/baf852d8-e7d9-444d-b764-da1c7de8a829)

These ones are fine, because they're filled:
![image](https://github.com/brave/leo/assets/7678024/1d23619b-c08b-4cfe-9cd1-7604ab26147f)

Close button on the customize dialog:
![image](https://github.com/brave/leo/assets/7678024/4e9cdba7-9289-4650-9a74-d36eb32045e8)

**Comment by fallaciousreasoning** [comment]
Also, not sure if it's related but this button looks like it has too much padding to me:
![image](https://github.com/brave/leo/assets/7678024/95ea66a5-3225-432b-9fa9-289982b17b54)

Here's the same button in Beta:
![image](https://github.com/brave/leo/assets/7678024/7fb7d29c-3d6b-48d4-bb52-159b1e658953)


cc @aguscruiz

**Comment by aguscruiz** [comment]
Those close buttons would be good candidates for the padding and hoverless buttons, I think. Those variants that are icon ondly and don't have a background change. What do you think?

As for the padding on that "Join cube" button, yeah, I can see it. The padding on these normal buttons have been the subject of long debates until we reached this final version. But it may be time to adjust a bit

**Comment by aguscruiz** [comment]
That close button though, yeah I think we should use the "icon-only" variant to solve it. But that's just me

**Comment by AlanBreck** [comment]
@fallaciousreasoning, from what I'm seeing, everything described above is to spec, according to the Figma doc.

I was going to say the same thing that @aguscruiz just said. All of the instances above where the hover background looks off seem like they ought to be icon-only buttons (or `fab`).

**Comment by fallaciousreasoning** [comment]
I think we need to be pretty careful about changing the buttons going forward - they're used in a lot of places, and it kinda defeats the point of having a design system if we need to go and audit all the places things are used when we update Nala.

**Comment by fallaciousreasoning** [comment]
Could we maybe get rid of the additional padding on the `plain` and `plain-faint` variants? It's definitely causing problems.

This is a `plain-faint` button. It used to align nicely with the edge but no longer does.
![image](https://github.com/brave/leo/assets/7678024/f13b2c44-d15a-4dd6-8bd5-810378a68aaa)

**Comment by AlanBreck** [comment]
I think another problem that this highlights is the need to deprecate `fab` and introduce switch to using a more logical name for icon only buttons. At least for the examples you mentioned, I think many wouldn't have caused an issue if they were icon only buttons, but `fab` just doesn't jump out as "oh yeah, that's what I should use in this case" when looking at Storybook.

**Comment by fallaciousreasoning** [comment]
One other thing I noticed is that the `filled` and `outline` buttons now have slightly different heights (I think because of the border). This makes a common use case (changing between the two states, depending on what the button is going to do) more complicated because the height changes

**Comment by AlanBreck** [comment]
> One other thing I noticed is that the `filled` and `outline` buttons now have slightly different heights (I think because of the border). This makes a common use case (changing between the two states, depending on what the button is going to do) more complicated because the height changes

Yep. I noticed the same thing. I believe it'll be a simple fix of adding a default border of `solid 1px transparent` to all buttons. Specifically changing this line: https://github.com/brave/leo/blob/7dc6a045947f82ff8200afb968b3757258534736/src/components/button/button.svelte#L145

```diff
- border: solid var(--border-width, 0px) var(--border-color, transparent);
+ border: solid var(--border-width, 1px) var(--border-color, transparent); 
```
